language-design.md

```diff
@@ -27,6 +43,7 @@ How do the themes of _Growing a Language_ relate to the lab we did this week?
 
 **Response**
 
+The Growing a Language essay was extremely relevant to the revamped cs5 lab we did in class this week. "A primitive is a word for which we can take it for granted that we all know what it means. For this talk, I chose to take as my primitives all the words of one syllable, and no more." [Steele, 1998] In Growing a Language the primatives were one syllable words. In the fluency lab, the primitives were these functions that were used to reverse, change volume etc. of sound clips. In growing a language Steele was able to "define" new words either word by word or by using a rule because the primitives that he started with were good tools to start with. In class, we rapidly identified a problem with the primitives in the cs5: they did not work well to define new words. The Growing a Language talk showed us how important it was to have a language that can grow and serve new purposes that the designer did not necessarily intend (or at least did not imagine) but the cs5 primitives did not allow for a user to do anything that was not built explicitly into the language. In the fluency lab we were tasked with making the cs5 primitives more similar to the one-syllable-english used by Steele in Growing a Language. If we could only make it so the primitives could be composed in order to serve new purposes, or allow a user to do a modify, save, or play action without doing them all at once then we would have updated the grammar to a point where a user could now use our primatives to define new words,rules and uses. This made us think of the rule set out for us in Bloch: "You can’t reasonably hope to imagine everything that everyone will do with an API, or how it will interact with every other part of a system." [Bloch,2006] In order to define a good API (or language) its necessary to realize that wont have thought of everything that a user will want to do so its vital that you write your grammar in such a way that new uses that you have not imagined can be incorporated.
 
 
```

Comment:
Hmm. Are you saying that using primitives is a theme? I would argue that you need more to make a theme. However I do like your point that the modify, save, and play functions are primitives in the new language we were trying to create. We didn't think of them that way.